tools/tscdocgen/main.go

```diff
@@ -1294,8 +1301,12 @@ func (e *emitter) createTypeLabel(t *typeDocType, indent int) string {
 					label += fmt.Sprintf("%s%s;\n",
 						strings.Repeat(" ", indent*4), e.createSignature(child.Signatures[0], decl, false))
 				} else {
+					var childType string
+					if child.Type != nil {
+						childType = e.createTypeLabel(child.Type, indent)
+					}
 					label += fmt.Sprintf("%s%s: %s;\n",
-						strings.Repeat(" ", indent*4), child.Name, e.createTypeLabel(child.Type, indent))
+						strings.Repeat(" ", indent*4), child.Name, childType)
```

Comment:
We're now hitting a new case that our doc generator hasn't run into before leading to dereferencing a nil pointer.

For now, I'm simply avoiding the nil reference, but it will require some work (which will have to be done subsequently), to generate the right signature. Right now it shows up as:

<img width="819" alt="Screen Shot 2019-11-26 at 1 24 10 PM" src="https://user-images.githubusercontent.com/710598/69673811-0a08d300-1050-11ea-9952-7223f5126761.png">

```typescript
validateTypedResource<TResource,TArgs>(resourceClass: {
    constructor: ;
}, validate: (props: Unwrap<NonNullable<TArgs>>, args: ResourceValidationArgs, reportViolation: ReportViolation) => Promise<void> | void): ResourceValidation
```

When it should look more like:

```typescript
function validateTypedResource<TResource extends Resource, TArgs>(resourceClass: {
    new(name: string, args: TArgs, ...rest: any[]): TResource
}, validate: (props: Unwrap<NonNullable<TArgs>>, args: ResourceValidationArgs, reportViolation: ReportViolation) => Promise<void> | void): ResourceValidation
```

I'll file a bug to follow-up and fix this. What we have now isn't ideal but not the end of the world.